utils/keys.go

```diff
@@ -0,0 +1,129 @@
+package utils
+
+import (
+	"encoding/pem"
+	"github.com/Sirupsen/logrus"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+	"sort"
+)
+
+// Exporter is a simple interface for the two functions we need from the Storage interface
+type Exporter interface {
+	Get(string) ([]byte, error)
+	ListFiles() []string
+}
+
+// Importer is a simple interface for the one function we need from the Storage interface
+type Importer interface {
+	Set(string, []byte) error
+}
+
+// ExportKeysByGUN exports all keys filtered to a GUN
+func ExportKeysByGUN(to io.Writer, s Exporter, gun string) error {
+	keys := s.ListFiles()
+	sort.Strings(keys) // ensure consistenct. ListFiles has no order guarantee
+	for _, k := range keys {
+		dir := filepath.Dir(k)
+		if dir == gun { // must be full GUN match
+			if err := ExportKeys(to, s, k); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// ExportKeysByID exports all keys matching the given ID
+func ExportKeysByID(to io.Writer, s Exporter, ids []string) error {
+	want := make(map[string]struct{})
+	for _, id := range ids {
+		want[id] = struct{}{}
+	}
+	keys := s.ListFiles()
+	for _, k := range keys {
+		id := filepath.Base(k)
+		if _, ok := want[id]; ok {
+			if err := ExportKeys(to, s, k); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// ExportKeys copies a key from the store to the io.Writer
+func ExportKeys(to io.Writer, s Exporter, from string) error {
+	// get PEM block
+	k, err := s.Get(from)
+	if err != nil {
+		return err
+	}
+
+	// parse PEM blocks if there are more than one
+	for block, rest := pem.Decode(k); block != nil; block, rest = pem.Decode(rest) {
+		// add from path in a header for later import
+		block.Headers["path"] = from
```

Comment:
Non-blocking - since there seems to be no objection yet to #838, should we change this to GUN in this PR or are we waiting to do that in a future PR?